src/framework/filterControl/forms/MixedTypeValueFilterForm.jsx

```diff
@@ -0,0 +1,137 @@
+
+import PropTypes from 'prop-types';
+import React, {
+  Component,
+} from 'react';
+
+import {
+  Filter,
+  FilterOption,
+} from '../../services';
+
+import CheckBox from '../../checkBox/CheckBox.jsx';
+import MixedTypeValueFilter from '../../services/filter/MixedTypeValueFilter';
+import PrimaryButton from '../../button/PrimaryButton.jsx';
+
+export default class MixedTypeValueFilterForm extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.onCheckBoxClick = this.onCheckBoxClick.bind(this);
+    this.onClickAddButton = this.onClickAddButton.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+
+    this.options = props.filterOption.comparisonParameters.options;
+
+    let selectedOptions;
+    let inputValue;
+    if (this.props.comparisonValue) {
+      selectedOptions = this.options.map(option =>
+        Boolean(props.comparisonValue.discreteValues.find(
+          element => element.value === option.value)
+        )
+      );
+      inputValue = this.props.comparisonValue.inputValue;
+    } else {
+      selectedOptions = (new Array(this.options.length)).fill(false);
+      inputValue = '';
+    }
+
+    this.state = {
+      inputValue,
+      selectedOptions,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.comparisonValue) {
+      if (!this.deepEqualsProps(
+          this.props.comparisonValue, prevProps.comparisonValue)) {
+        // eslint-disable-next-line react/no-did-update-set-state
```

Comment:
Skipping eslint rules is usually hacky or bad... there is usually a better way to do it.

Reply:
This is actually due to having 2 sources of truth....props, and state...

Do we need the `inputValue` in the state?